### DISCUSSION TITLE
csurf middleware and nuxtjs-alt/http

### DISCUSSION THREAD

**Discussion by mittci** [nuxt/nuxt#16442] [Questions] [answered]
@danielroe @pi0 @manniL 
What is wrong with that https://github.com/Teranode/nuxt-module-alternatives/issues/111

The `$http.$post('/api/me', { body: { email: '[test@test.com](mailto:test@test.com)', pass: '123' } })` result (or `$http.$post('/api/me', { email: '[test@test.com](mailto:test@test.com)', pass: '123' }})` )

```
Response header

set-cookie: XSRF-TOKEN=SqMGdF8Z-vYDoN4lglcb3HFfRp43g6q-bVUQ; Path=/
set-cookie: xsrf-token=SqMGdF8Z-vYDoN4lglcb3HFfRp43g6q-bVUQ; Path=/


Request header

x-xsrf-token: tNRlO9qn-u4Ulf7KfjsUWb6N-IisEn7JPFng

```

But when useNuxtApp().$auth.loginWith('local', { body: { email: 'mail@mail.com', password: '123qwe' }}) result is
```
Status Code: 403 Forbidden

Response header - no set cookies

Request header - no x-xsrf-token
```

If call first api when `$auth.loginWith` work well? but if call it first get 403 error (the same 403 error get if make direct api call after)

**Comment by danielroe** [comment]
Please do not ping maintainers like this, and allow the maintainer of that repository some time to actually look at the issue.

**Comment by mittci** [comment, marked as answer]
In @refactorjs/ofetch is `config.headers[config.xsrfHeaderName] = decodeURIComponent(cookie);` but need to be `config.headers.set(config.xsrfHeaderName, decodeURIComponent(cookie));` i think?

And like in ohmyfetch `config.headers = new Headers(config.headers)` before

That will be

```
addXSRFHeader_fn = function(config) {
  const cookie = getCookie(config.xsrfCookieName);
  const cookies = getCookies();

  if (config.credentials === "include" && config.xsrfCookieName && cookies[config.xsrfCookieName]) {
    config.headers[config.xsrfHeaderName] = decodeURIComponent(cookie);
  }
  return config;
};
```


Like so

```
addXSRFHeader_fn = function(config) {
  const cookie = getCookie(config.xsrfCookieName);
  const cookies = getCookies();

  if (config.credentials === "include" && config.xsrfCookieName && cookies[config.xsrfCookieName]) {
    console.log('addXSRFHeader_fn', config.headers)

    if (config.headers.constructor.name === 'Object') {
      config.headers = new Headers(config.headers)
    }

    // config.headers[config.xsrfHeaderName] = decodeURIComponent(cookie);
    config.headers.set(config.xsrfHeaderName, decodeURIComponent(cookie));
  }
  return config;
};
```
